gcovr/utils.py

```diff
@@ -380,15 +393,18 @@ def presentable_filename(filename, root_filter):
 
 
 def fixup_percent(percent):
-    # output csv percent values in range [0,1.0]
-    return percent / 100 if percent is not None else None
+    # output percent values in range [0,1.0]
+    return max(min(1.0, percent / 100), 0.0) if percent is not None else None
```

Comment:
That is a good point. I think the best is to calculate the percent values when writing thew output files instead of writing them to an intermediate file.
There is als a problem in the JSON intermediate format. The `branch_percent` is from 0% to 100% and the `line_percent` from 0.0 to 1.0. I think this should be fixed (maybe a separate PR) but all solutions will be incompatible:
- Use 0% to 100% -> `line_percent` will be incompatible.
- Use 0.0 to 1.0 -> `branch_percent` will be incompatible.
- Remove the values -> also incompatible.

I prefer the last solution because the merging of data doesn't need to merge the percent values and the FP arithmetic is removed from the data model.

Reply:
Breaking changes are acceptable for bugfixes, and clearly inconsistent behaviour might be a bug.

For the summary report, it is not necessary to restrict ourselves to necessary info – showing percentages could be helpful, even though technically redundant. Unfortunately, the percentage format looks like this:

| level | branch | lines |
|---|---|---|
| per-file | 0..1 | 0..1 |
| global | 0..100 | 0..100 |

I.e. line and branch percentages are consistent with each other on the same level, but the levels are inconsistent. I'd leave it at that, though a “percentage” should really go from 0..100.